cloud/deployment/fromfile/fromfile.go

```diff
@@ -158,189 +148,416 @@ func CreateOrUpdate(inputFile, action string, client astro.Client, coreClient as
-		err            error
+		defaultOptions          astroplatformcore.WorkerQueueOptions
+		configOptions           astroplatformcore.DeploymentOptions
+		listQueues              []astroplatformcore.WorkerQueue
+		astroMachine            astroplatformcore.WorkerMachine
+		listQueuesRequest       []astroplatformcore.WorkerQueueRequest
+		listHybridQueuesRequest []astroplatformcore.HybridWorkerQueueRequest
+		err                     error
 	)
+	deploymentType := transformDeploymentType(deploymentFromFile.Deployment.Configuration.DeploymentType)
 
 	// Add worker queues if they were requested
 	if hasQueues(deploymentFromFile) {
 		// transform inspect.WorkerQ to []astro.WorkerQueue
-		listQueues, err = getQueues(deploymentFromFile, nodePools, existingDeployment.WorkerQueues)
+		if existingDeployment.WorkerQueues != nil {
+			listQueues = *existingDeployment.WorkerQueues
+		}
+		listQueues, err = getQueues(deploymentFromFile, nodePools, listQueues)
+		if err != nil {
+			return err
+		}
+		// get defaults for min-count, max-count and concurrency from API
+		deploymentOptions, err := deployment.GetPlatformDeploymentOptions("", astroplatformcore.GetDeploymentOptionsParams{}, astroPlatformCore)
 		if err != nil {
-			return astro.CreateDeploymentInput{}, astro.UpdateDeploymentInput{}, err
+			return err
 		}
-		if deploymentFromFile.Deployment.Configuration.Executor == deployment.CeleryExecutor {
-			// get defaults for min-count, max-count and concurrency from API
-			defaultOptions, err = workerqueue.GetWorkerQueueDefaultOptions(client)
-			if err != nil {
-				return astro.CreateDeploymentInput{}, astro.UpdateDeploymentInput{}, err
-			}
-			if deployment.IsDeploymentHosted(deploymentFromFile.Deployment.Configuration.DeploymentType) || deployment.IsDeploymentDedicated(deploymentFromFile.Deployment.Configuration.DeploymentType) {
-				configOptions, err = client.GetDeploymentConfig()
-				if err != nil {
-					return astro.CreateDeploymentInput{}, astro.UpdateDeploymentInput{}, err
+
+		defaultOptions = deploymentOptions.WorkerQueues
+		configOptions = deploymentOptions
+		for i := range listQueues {
+			if deployment.IsDeploymentStandard(deploymentType) || deployment.IsDeploymentDedicated(deploymentType) {
+				astroMachines := configOptions.WorkerMachines
+				for j := range astroMachines {
+					if astroMachines[j].Name == *listQueues[i].AstroMachine {
+						astroMachine = astroMachines[j]
+					}
 				}
-			}
-			for i := range listQueues {
+				var workerQueue astroplatformcore.WorkerQueueRequest
+				workerQueue.Id = &listQueues[i].Id
+				workerQueue.Name = listQueues[i].Name
+				workerQueue.IsDefault = listQueues[i].IsDefault
+				workerQueue.MaxWorkerCount = listQueues[i].MaxWorkerCount
+				workerQueue.MinWorkerCount = listQueues[i].MinWorkerCount
+				workerQueue.WorkerConcurrency = listQueues[i].WorkerConcurrency
+				workerQueue.AstroMachine = astroplatformcore.WorkerQueueRequestAstroMachine(*listQueues[i].AstroMachine)
```

Comment:
does this work instead?
```suggestion
				workerQueue.AstroMachine = astroplatformcore.WorkerQueueRequestAstroMachine(listQueues[i].AstroMachine)
```

Reply:
astroplatformcore.WorkerQueueRequestAstroMachine needs a string and listQueues[i].AstroMachine is pointer